arlo_server/validation.py

```diff
@@ -0,0 +1,15 @@
+from werkzeug.exceptions import BadRequest
+
+
+def check_required_fields(json, fields, prefix=""):
```

Comment:
I looked into something like this a while ago. Other than using something like GraphQL, something like this might be best: https://json-schema.org. I wouldn't say it's required for this, but that's the direction I think we should go.

Reply:
I've used json schema in the past and it worked well. Think it's worth setting up now?

Reply:
Ooh I've never used GraphQL before but it looks cool too...

Reply:
If you feel you can set up JSON schema without too much fuss, I’d love to see it. I did a spike with GraphQL and personally really liked it. That opinion was not universally shared.

Reply:
that would be my fault for pushing back on it :) From everyhing I've seen, graphql adds can rapidly complicate an API surface, and I tend to lean hard towards keeping API contracts as simple as possible.

I'm all for schema validation using JSON schema.